src/nupic/types/Fraction.cpp

```diff
@@ -366,7 +366,7 @@ namespace nupic
 
     while(components.size())
     {
-      std::swap<int>(numerator_, denominator_);
+      std::swap(numerator_, denominator_);
```

Comment:
> Why did you remove template param?

This was presenting problems at first try on port Nupic to Windows. Maybe with the new cross-platform workaround flags introduced into CMake and other compatibility improvements brought by VS 2015 and many other hacks will can be removed. :smiley: